include/alpaka/core/RuntimeMacros.hpp

```diff
@@ -0,0 +1,48 @@
+//! depending on the Acc. The std::runtime_error exception can be catched in the catch block.
+//!
+//! CUDA __trap function triggers std::runtime_error but can be catched during wait not exec.
+//!
+//! The OpenMP specification mandates that exceptions thrown by some thread must be handled by the same thread.
+//! Therefore std::runtime_error thrown by ALPAKA_THROW_ACC aborts the the program for OpenMP backends. If needed
+//! the SIGABRT signal can be catched by signal handler.
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && defined(__CUDA_ARCH__)
+#    define ALPAKA_THROW_ACC(MSG)                                                                                     \
+        {                                                                                                             \
+            printf(                                                                                                   \
+                "alpaka encountered a user-defined error condition while running on the CUDA back-end:\n%s",          \
+                (MSG));                                                                                               \
+            __trap();                                                                                                 \
+        }
+#elif defined(ALPAKA_ACC_GPU_HIP_ENABLED) && defined(__HIP_DEVICE_COMPILE__)
+#    define ALPAKA_THROW_ACC(MSG)                                                                                     \
+        {                                                                                                             \
+            printf(                                                                                                   \
+                "alpaka encountered a user-defined error condition while running on the HIP back-end:\n%s",           \
+                (MSG));                                                                                               \
+            abort();                                                                                                  \
+        }
+#elif defined(ALPAKA_ACC_SYCL_ENABLED) && defined(__SYCL_DEVICE_ONLY__)
+#    define ALPAKA_THROW_ACC(MSG)                                                                                     \
+        {                                                                                                             \
+            printf(                                                                                                   \
+                "alpaka encountered a user-defined error condition while running on the SYCL back-end:\n%s",          \
+                (MSG));                                                                                               \
+            assert(false);                                                                                            \
```

Comment:
Note: assert is a noop incase `-DNDEBUG` is set.

Reply:
We should use `std::abort` even if it is not ending the application similar to HIP.

Reply:
My tests previously done had generated the same error:

https://github.com/alpaka-group/alpaka/pull/2283#issuecomment-2169733670

Reply:
assert(false) works as I tested before:
```
spack load intel-oneapi-compilers@2023.1.0
$ icpx -fsycl test_abort.cpp
$ ./a.out 
AssertHandler::printMessage
test_abort.cpp:27: auto main()::(anonymous class)::operator()(sycl::handler &)::(anonymous class)::operator()() const: global id: [0,0,0], local id: [0,0,0] Assertion `false` failed.
Aborted (core dumped)
```

Reply:
It probably does not work if the user defines `NDEBUG`.

Can you try compiling with `-DNDEBUG` ?

Reply:
yes, when compiled with -DNDEBUG it does not abort`  icpx -fsycl test_abort.cpp -DNDEBUG`